pkg/repository/backup_repo_op.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright The Velero Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/label"
+)
+
+// A BackupRepositoryKey uniquely identify a backup repository
+type BackupRepositoryKey struct {
+	VolumeNamespace string
+	BackupLocation  string
+	RepositoryType  string
+}
+
+const (
+	notFoundError = "not found BackupRepository"
+)
+
+func repoLabelsFromKey(key BackupRepositoryKey) labels.Set {
+	return map[string]string{
+		velerov1api.VolumeNamespaceLabel: label.GetValidName(key.VolumeNamespace),
+		velerov1api.StorageLocationLabel: label.GetValidName(key.BackupLocation),
+		velerov1api.RepositoryTypeLabel:  label.GetValidName(key.RepositoryType),
+	}
+}
+
+// GetBackupRepository gets a backup repository through BackupRepositoryKey and ensure ready if required.
+func GetBackupRepository(ctx context.Context, cli client.Client, namespace string, key BackupRepositoryKey, ensureReady bool) (*velerov1api.BackupRepository, error) {
```

Comment:
If 99% of the time this func will be called with ensureReady==true, would it be better to replace it with `skipReadyCheck` and make it an optional parm?  Such that the caller will not need to remember setting the last parm.

Reply:
Done, changed the `ensureReady` to an optional param.